Move ext/common to runtime/common (Pkgs 8/N)

padentomasello opened the pull request:
**IMPORTANT: Please do not create a Pull Request without creating an issue first.** Changes *must* be discussed.

**Original Issue**: [corresponding issue on Github]

*Note:* You can add `closes #[issue number]` to automatically close the issue that this PR resolves when it is merged.

### Summary
[Explain the details for this change and the problem that the pull request solves]

### Test Plan (required)
[steps by which you tested that your fix resolves the issue. These might include specific commands and configurations]

Commit:
If it's common, probably shouldn't be in ext...

jacobkahn commented on CMakeLists.txt:
```diff
@@ -233,13 +233,6 @@ if (FL_BUILD_CORE)
     )
-
-  # ------------------------ Extensions ------------------------
-  set(FL_EXT_DIR "${FL_ROOT_DIR}/ext")
```
🎉

jacobkahn reviewed (changes requested):
I'm a bit torn on `SequentialBuilder` and `DistributedUtils` being in `fl`. The archfile abstraction is thoroughly a runtime-only thing and should probably be in `fl_pkg_runtime`. I'm leaning the same way for `DistributedUtils` considering it wraps the `distributedInit` function from `fl/distributed` with something called `initDistributed` and is really only a runtime convenience function.

Is the path rename for `BuildGoogleTest.cmake` correct? Seems like that was moved to `ext/test`... which might be empty now? If this is fixed in a later change, ignore this.

jacobkahn commented:
One other issue this introduces - since `Serializer.h` is moved to the core, it introduces a core dependency in `lib` since, that file depends on `lib` symbols. It should also probably be in `fb_pkg_runtime`, since like the other things I"ve mentioned above, it's convenience wrappers for things in the core.

Commit:
Revert "If it's common, probably shouldn't be in ext..."

This reverts commit 3ee51c9ff92ff517a171b61baf84a4dd6c877744.

Commit:
Move ext/common to runtime/common

Commit:
PR fixes

padentomasello commented:
@jacobkahn I agree with you. That was my initial thought too, but there are some dependencies on DistributedUtils in the tests for speech, and I didn't want to add a dependency on the runtime pkg from the speech pkg. For now, I moved everything into runtime, and made only the tests in speech depenedent on the runtime pkg. Added a todo to revit.
